agent/consul/gateway_locator_test.go

```diff
@@ -422,6 +479,16 @@ type testServerDelegate struct {
 
 	isLeader    bool
 	lastContact time.Time
+
+	dcSupportsFederationStates int32 // atomic
```

Comment:
This needs to be 64 byte aligned to not panic on 32 bit builds right? The bool above will break that I think. Probably safest to move it to the top of the struct with the usual "watch out for alignment" comment?

Reply:
I missed this was on a test only. Still would be nice if we could run tests on all our supported platforms (e.g. arm32) without a panic but less important.

Reply:
Moved it.